File: typescript/src/data_models/external_task/results/external_task_success_result.ts
import {ExternalTaskResultBase} from './external_task_result_base';

/**
 * Contains the result set for an ExternalTask that finished successfully.
 */
export class ExternalTaskSuccessResult<TResult> extends ExternalTaskResultBase {

  public readonly result: TResult;

  constructor(externalTaskId: string, result: TResult) {
    super(externalTaskId);
    this.result = result;
  }

}

File: dotnet/src/DataModels/ExternalTask/Results/ExternalTaskSuccessResult.cs
namespace ProcessEngine.ConsumerAPI.Contracts.DataModel
{
    /// <summary>
    /// Contains the result set for an ExternalTask that finished successfully.
    /// </summary>
    public class ExternalTaskSuccessResult<TResult> : ExternalTaskResultBase
    {
        /// <summary></summary>
        /// <param name="externalTaskId"></param>
        /// <param name="result"></param>
        public ExternalTaskSuccessResult(string externalTaskId, TResult result)
            :base(externalTaskId)
        {
            this.Result = result;
        }

        /// <summary>
        /// The result with which the ExternalTask has finished.
        /// </summary>
        /// <value></value>
        public TResult Result { get; private set; }
    }
}
